packages/engine/Specs/Scene/Model/RuntimeModelInstancingPipelineStageSpec.js

```diff
@@ -0,0 +1,289 @@
+        ]);
+
+        expect(runtimeNode.instancingTransformsBuffer).toBeDefined();
+        // The resource will be counted by NodeStatisticsPipelineStage.
+        expect(renderResources.model.statistics.geometryByteLength).toBe(0);
+      });
+    });
+
+    it("correctly creates transform matrices", function () {
+      return loadGltf(sampleGltfUrl).then(function (gltfLoader) {
+        const components = gltfLoader.components;
+        const node = components.nodes[0];
+        const renderResources = mockRenderResources(node);
+        const modelInstances = renderResources.model.sceneGraph.modelInstances;
+
+        const expectedTransformsTypedArray = new Float32Array([
+          -0.410076379776001, 0.7071067690849304, 0.576053261756897, 0,
+          -0.410076379776001, -0.7071067690849304, 0.576053261756897, 0,
+          0.8146623373031616, 0, 0.5799355506896973, 0, 0, 0, 0, 10, 10, 10,
+          -0.410076379776001, 0.7071067690849304, 0.576053261756897, 0,
+          -0.410076379776001, -0.7071067690849304, 0.576053261756897, 0,
+          0.8146623373031616, 0, 0.5799355506896973, 0, 0, 0, 0, 20, 20, 20,
+        ]);
+        const transformsTypedArray =
+          RuntimeModelInstancingPipelineStage._getTransformsTypedArray(
+            modelInstances,
+          );
+
+        expect(transformsTypedArray.length).toEqual(
+          expectedTransformsTypedArray.length,
+        );
+        for (let i = 0; i < expectedTransformsTypedArray.length; i++) {
+          expect(transformsTypedArray[i]).toEqualEpsilon(
+            expectedTransformsTypedArray[i],
+            CesiumMath.EPSILON10,
+          );
+        }
+      });
+    });
+
+    it("model instances update stage updates transform vertex attributes", function () {
+      return loadGltf(sampleGltfUrl).then(function (gltfLoader) {
+        const components = gltfLoader.components;
+        const node = components.nodes[0];
+        const renderResources = mockRenderResources(node);
+        const runtimeNode = renderResources.runtimeNode;
+        const sceneGraph = renderResources.model.sceneGraph;
+
+        scene.renderForSpecs();
+        RuntimeModelInstancingPipelineStage.process(
+          renderResources,
+          node,
+          scene.frameState,
+        );
+
+        const context = scene.frameState.context;
+        const usage = BufferUsage.STATIC_DRAW;
+
+        const expectedTransformsTypedArray = new Float32Array([
+          -0.410076379776001, 0.7071067690849304, 0.576053261756897, 0,
+          -0.410076379776001, -0.7071067690849304, 0.576053261756897, 0,
+          0.8146623373031616, 0, 0.5799355506896973, 0, 0, 0, 0, 10, 10, 10,
+          -0.410076379776001, 0.7071067690849304, 0.576053261756897, 0,
+          -0.410076379776001, -0.7071067690849304, 0.576053261756897, 0,
+          0.8146623373031616, 0, 0.5799355506896973, 0, 0, 0, 0, 20, 20, 20,
+        ]);
+
+        const expectedTransformsBuffer = Buffer.createVertexBuffer({
```

Comment:
It looks like this test is jumping through a few hoops to validate the buffer contents. 

Instead of creating a new buffer, perhaps consider validating the output of `RuntimeModelInstancingPipelineStage._getTransformsTypedArray` against `expectedTransformsTypedArray`, and then checking `runtimeNode.instancingTransformsBuffer`'s `usage` and `byteLength` properties directly.